Update: Use offsite navigation editor on the navigation inspector sidebar.

jorgefilipecosta opened the pull request:
This PR updates the navigation inspector sidebar to use the new navigation offsite editor being worked on:
https://github.com/orgs/WordPress/projects/60/views/1

If the navigation offsite editor is disabled, the edit button and the appended are not shown, so it behaves similarly to what we have now.
This change also allows browse mode navigation menus to take advantage of the new menu editor. After this PR, and https://github.com/WordPress/gutenberg/pull/46436 are merged.

cc: @scruffian 
## Screenshots or screencast <!-- if applicable -->
Off-canvas editor enable:
![image](https://user-images.githubusercontent.com/11271197/206756396-a8f2f093-da00-43cc-901c-1cdc769a6a74.png)

Off-canvas editor disable: 
![image](https://user-images.githubusercontent.com/11271197/206756434-824c58cc-beee-489d-9b4e-0046e9f1c161.png)

jorgefilipecosta committed:
Update: Use offsite navigation editor component on the navigation inspector.

scruffian commented:
This looks like a good start. I noticed quite a few things don't work:
1. The block inserter throws an error
2. The "Add a submenu item" option doesn't work
3. The edit icon doesn't open the editing controls.

jorgefilipecosta commented:
> This looks like a good start. I noticed quite a few things don't work:
> 
> 1. The block inserter throws an error
> 2. The "Add a submenu item" option doesn't work
> 3. The edit icon doesn't open the editing controls.

Hi @scruffian, the inserter edit, etc, are hidden without the experiment enabled. I guess we can merge as these features are hidden without the experiment and then look into fixing these issues in the OffCanvasEditor component?

scruffian commented:
> Hi @scruffian, the inserter edit, etc, are hidden without the experiment enabled. I guess we can merge as these features are hidden without the experiment and then look into fixing these issues in the OffCanvasEditor component?

That's true, but if we're going down this route we should probably put the <OffCanvasEditor> behind the same experimental flag, so it only shows up when people opt in.

jorgefilipecosta commented:
> That's true, but if we're going down this route we should probably put the behind the same experimental flag, so it only shows up when people opt-in
.
The edit and inserter etc are behind a feature flag (the same one that the offsite navigation editor uses) the rest works on the navigation inspector exactly as we have now.

scruffian commented:
There are other things that the Off Canvas Editor experiment enables though for example "Add a submenu item"
<img width="315" alt="Screenshot 2022-12-14 at 13 10 15" src="https://user-images.githubusercontent.com/275961/207604115-4c33b64e-4f1c-4ac3-9648-4889cd64de98.png">

I just wonder if its safer to just default to the old view until you enable the experiment...

getdave reviewed (changes requested):
With respect, I really think we should have everything related to the offcanvas editor behind the experiment flag.

Why?

The engineering decisions behind that offcanvas have been taken to prioritise expediency to allow for early testing and UX feedback. As a result it is not suitable for exposure outside of the experiment. 

Doing that at this time could hinder progress on the offcanvas experiment as bugs would need to be fixed and `trunk` more closely monitored for breakages. At this time we're still waiting on feedback from the community as to whether offcanvas is a good experience.

If that makes the `Navigation` element of the Navigation panel redundant then it should be removed for now or again be placed behind the experiment flag.

jorgefilipecosta commented:
> With respect, I really think we should have everything related to the offcanvas editor behind the experiment flag.
> 
> Why?
> 
> The engineering decisions behind that offcanvas have been taken to prioritise expediency to allow for early testing and UX feedback. As a result it is not suitable for exposure outside of the experiment.
> 
> Doing that at this time could hinder progress on the offcanvas experiment as bugs would need to be fixed and `trunk` more closely monitored for breakages. At this time we're still waiting on feedback from the community as to whether offcanvas is a good experience.
> 
> If that makes the `Navigation` element of the Navigation panel redundant then it should be removed for now or again be placed behind the experiment flag.

Hi @getdave, thank you for sharing your thoughts. I updated this PR to use the new component if the flag is enabled. On the browse sidebar, we will also only show the navigation menu item if the experiment is enabled. I think this addresses your concerns.

scruffian reviewed (approved):
Thank you!

getdave commented:
@jorgefilipecosta Thank you for being so considerate and apologies for blocking with that review 🙇
